### PR TITLE
Add libcramjam

### PR DESCRIPTION
**Pull request by milesgranger**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by milesgranger** [comment]
@conda-forge/staged-recipes, ready for review. 👌

**Review by mfansler — COMMENTED** [review]
*(Soft review from not a Rust expert)*

The header only seems to expose snappy. Is this still in development?

Also, I see everything is checked off, but this is statically redistributing several libraries. One should include additional licenses for these if they don't all fall under BSD3. E.g., [`zstd` distributes with MIT](https://github.com/gyscos/zstd-rs/blob/main/LICENSE).

**Comment by milesgranger** [comment]
Thanks @mfansler!

> The header only seems to expose snappy. Is this still in development?

This portion is, yes. Just starting off with exposed C functions but having on conda-forge will ease later development. 

> One should include additional licenses for these if they don't all fall under BSD3.

Sorry, maybe I misunderstood, but I thought that was what the `THIRDPARTY.yml` generation was doing, no?

**Comment by mfansler** [comment]
> > One should include additional licenses for these if they don't all fall under BSD3.
> 
> Sorry, maybe I misunderstood, but I thought that was what the `THIRDPARTY.yml` generation was doing, no?

You're right - all good!

**Review by mfansler — APPROVED** [review]
This looks good to me - verified artifacts, reasonable tests, logs all passing!

However, I'm not a regular in the Rust review pool, so I'll defer to someone more experienced to merge.

**Comment by milesgranger** [comment]
Great, thanks for the initial look and feedback! :+1:

File: recipes/libcramjam/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,52 @@
+# Note: there are many handy hints in comments in this example -- remove them when you've finalized your recipe
+# If your package is python based, we recommend using Grayskull to generate it instead:
+# https://github.com/conda-incubator/grayskull
+
+# Jinja variables help maintain the recipe as you'll update the version only here.
+# Using the name variable with the URL in line 16 is convenient
+# when copying and pasting from another recipe, but not really needed.
```
Comment (xhochy) [inline recipes/libcramjam/meta.yaml:7]:
```suggestion
```

File: recipes/libcramjam/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    - {{ compiler('m2w64_c') }}  # [win]
+    - pkg-config                 # [unix]
+    - cargo-bundle-licenses
+    - cargo-c
+
+test:
+  commands:
+    - test -e $PREFIX/include/cramjam/cramjam.h              # [unix]
+    - test -e $PREFIX/lib/libcramjam.so                      # [linux]
+    - test -e $PREFIX/lib/libcramjam.dylib                   # [osx]
```
Comment (xhochy) [inline recipes/libcramjam/meta.yaml:37]:
```suggestion
    - test -e $PREFIX/lib/libcramjam${SHLIB_EXT}                      # [unix]
```

**Review by xhochy — CHANGES_REQUESTED** [review]

**Review by xhochy — APPROVED** [review]

**Pull request merged**